infrastructure/ssz/src/test/java/tech/pegasys/teku/infrastructure/ssz/schema/SszSchemaTestBase.java

```diff
@@ -45,6 +45,10 @@ void getDefaultTree_shouldBeEqualToDefaultStructure(SszSchema<SszData> schema) {
   @MethodSource("testSchemaArguments")
   @ParameterizedTest
   void sszDeserialize_tooLongSszShouldFailFastWithoutReadingWholeInput(SszSchema<SszData> schema) {
+    if (schema instanceof SszOptionalSchema<?, ?>) {
+      // empty SszOptional couldn't pass this test
```

Comment:
Not sure why the SszOptional is the special case here?

Reply:
The test adds to the reader buffer some random byte sequence. It expects you to break up reading buffer before consuming any of the added data. 1 more byte read - failed. Empty serialized SszOptional is byte[0]. So we should somehow guess that we have empty SszOptional with non-zero bytes buffer provided without reading any extra of correctly serialized (0) bytes, which is impossible for this type with EIP-6475 proposed  serialization.

Reply:
Oh, got it 👍